bench/word_count/word_count.cpp

```diff
@@ -22,18 +21,44 @@ int main(int argc, char *argv[]) {
   }
 
   ifstream file(argv[1]);
-  if (!file.is_open()) {
+  
+  if (file)
+  {
+    std::string str;
+    
+    file.seekg(0, std::ios::end);
+    str.reserve(file.tellg());
+    file.seekg(0, std::ios::beg);
+
+    str.assign((std::istreambuf_iterator<char>(file)),
```

Comment:
I believe `file.read(str.data(), str.size());` is faster :)

Reply:
Agreed, but technically you should resize the std::string rather than call reserve. I wasn't caring too much about file performance, it was the bad use of constantly allocating strings that I was fixing.